substrate/frame/revive/src/vm/evm/instructions/host.rs

```diff
@@ -178,17 +157,24 @@ pub fn blockhash<'ext, E: Ext>(context: Context<'_, 'ext, E>) {
 /// Loads a word from storage.
 pub fn sload<'ext, E: Ext>(context: Context<'_, 'ext, E>) {
 	popn_top!([], index, context.interpreter);
-
-	let Some(value) = context.host.sload(context.interpreter.input.target_address(), *index) else {
-		context.interpreter.halt(InstructionResult::FatalExternalError);
-		return;
+	// NB: SLOAD loads 32 bytes from storage (i.e. U256).
+	gas!(context.interpreter, RuntimeCosts::GetStorage(32));
+	let key = Key::Fix(index.to_be_bytes());
+	let value = context.interpreter.extend.get_storage(&key);
+
+	*index = if let Some(storage_value) = value {
+		if storage_value.len() != 32 {
+			// sload always reads a word
+			context.interpreter.halt(InstructionResult::FatalExternalError);
+			return;
+		}
+		let mut bytes = [0u8; 32];
+		bytes.copy_from_slice(&storage_value);
```

Comment:
Bikeshedding
```suggestion
		// sload always reads a word
		let Ok::<[u8; 32], _>(bytes) = storage_value.try_into() else {
			context.interpreter.halt(InstructionResult::FatalExternalError);
			return
		};
```

Reply:
I think he just wants to bike-shed this into the more readable form he suggested.